src/command_buffer.rs

```diff
@@ -0,0 +1,221 @@
+    /// recorder.run_on(&mut world);
+    /// assert!(world.contains(a));
+    /// ```
+    pub fn run_on(&mut self, world: &mut World) {
+        let mut mark: usize = 0;
+        self.sort_buffered();
+        (0..self.ent.len()).for_each(|_| {
+            let (ent, comps) = self.build(mark);
+            world.spawn_at(ent, comps);
+            mark += 1;
+        });
+        self.clear();
+    }
+
+    pub(crate) fn sort_buffered(&mut self) {
+        for index in self.ent.iter() {
+            self.info[index.beg..index.end].sort_unstable_by_key(|z| z.ty_info);
+        }
+    }
+
+    pub(crate) fn build(&mut self, mark: usize) -> (Entity, ReadyBuffer<'_>) {
+        let end = self.ent[0].end - self.ent[0].beg;
+        self.ids
+            .extend(self.info[0..end].iter().map(|x| x.ty_info.id()));
+        let ent = self.ent[mark].entity;
+        (ent, ReadyBuffer { buffer: self })
+    }
+
+    /// Drop previously `recorded` entities and their components
+    ///
+    /// Recorder is cleared implicitly when entities are spawned, so usually this doesn't need to
+    /// be called
+    pub fn clear(&mut self) {
+        self.ids.clear();
+        self.ent.clear();
+        self.cursor = 0;
+        unsafe {
+            for info in self.info.drain(..) {
+                info.ty_info.drop(self.storage.as_ptr().add(info.size));
+            }
+        }
+    }
+}
+
+unsafe impl Send for CommandBuffer {}
+unsafe impl Sync for CommandBuffer {}
+
+impl Drop for CommandBuffer {
+    fn drop(&mut self) {
+        self.clear();
+        if self.layout.size() != 0 {
+            unsafe {
+                dealloc(self.storage.as_ptr(), self.layout);
+            }
+        }
+    }
+}
+
+impl Default for CommandBuffer {
+    /// Create an empty buffer
+    fn default() -> Self {
+        Self {
+            ent: Vec::new(),
+            storage: NonNull::dangling(),
+            layout: Layout::from_size_align(0, 8).unwrap(),
+            cursor: 0,
+            info: Vec::new(),
+            ids: Vec::new(),
+        }
+    }
+}
+
+/// The output of an '[CommandBuffer]` suitable for passing to
+/// [`World::spawn_into`](crate::World::spawn_into)
+pub struct ReadyBuffer<'a> {
+    buffer: &'a mut CommandBuffer,
+}
+unsafe impl DynamicBundle for ReadyBuffer<'_> {
+    fn with_ids<T>(&self, f: impl FnOnce(&[TypeId]) -> T) -> T {
+        f(&self.buffer.ids)
+    }
+
+    fn type_info(&self) -> Vec<TypeInfo> {
+        let end = self.buffer.ent[0].end - self.buffer.ent[0].beg;
+        self.buffer.info[0..end].iter().map(|x| x.ty_info).collect()
+    }
+
+    unsafe fn put(self, mut f: impl FnMut(*mut u8, TypeInfo)) {
+        let end = self.buffer.ent[0].end - self.buffer.ent[0].beg;
+        for info in self.buffer.info.drain(0..end) {
+            let ptr = self.buffer.storage.as_ptr().add(info.size);
+            f(ptr, info.ty_info);
+        }
+    }
+}
+
+/// Data required to store components and their offset  
+pub struct InfoBuffer {
```

Comment:
This structure is not a buffer.
```suggestion
pub struct ComponentInfo {
```